pkg/scheduler/core/generic_scheduler.go

```diff
@@ -444,12 +444,10 @@ func (g *genericScheduler) prioritizeNodes(
 	}
 
 	// Summarize all scores.
-	result := make(framework.NodeScoreList, 0, len(nodes))
-
-	for i := range nodes {
-		result = append(result, framework.NodeScore{Name: nodes[i].Name, Score: 0})
-		for j := range scoresMap {
-			result[i].Score += scoresMap[j][i].Score
+	result := make(framework.NodeScoreList, len(nodes), len(nodes))
+	for _, nodeScores := range scoresMap {
+		for node := range nodes {
+			result[node].Score += nodeScores[node].Score
```

Comment:
and the name? it probably needs to happen in a separate loop, otherwise you would be doing multiple copies.

Reply:
we can do: "if i == 0", then "result[i].Name = nodes[i].Name"

Anyway, I do think it's necessay to build a UT benchmark test to evaluate the performance.

Reply:
I think it's overkill. TBH, scoring the nodes is way more expensive.

But, this is an obvious complexity improvement. The original is O(n*m*lg(m)), plus we are kicking out of the smaller caches the slice for each plugin. The new code is O(n*m).

Reply:
> and the name? it probably needs to happen in a separate loop, otherwise you would be doing multiple copies.

Do you mean like a loop to initialize the `result` list with each node's name and a score of 0?

Reply:
I'll leave it up to you, but we need the name somewhere